### PR TITLE
test: Fix flakey SentryHttpTransport tests

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

The unit tests sometimes fail in the CI, because the request count in
SentryHttpTransportTests sometimes doesn't match. The TestRequestManager contained a non
thread safe array of the requests. The writing to this is array is now synchronized and this should
fix the flakey tests.

## :bulb: Motivation and Context

The unit tests in CI are sometimes failing, for example https://github.com/getsentry/sentry-cocoa/runs/1459801122?check_suite_focus=true

## :green_heart: How did you test it?
Before the change, I could reproduce the tests failing by running the tests around 10 to 20 times. After the change I wasn't able to reproduce the failing tests. The CI will tell if this change fixes all the flaky tests.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Tests/SentryTests/Networking/TestRequestManager.swift [hunk 1]
```diff
@@ -19,7 +21,9 @@ public class TestRequestManager: NSObject, RequestManager {
     var responseDelay = 0.0
     public func add( _ request: URLRequest, completionHandler: SentryRequestOperationFinished? = nil) {
         
+        semaphore.wait()
         requests.append(request)
```
Comment (marandaneto) [inline Tests/SentryTests/Networking/TestRequestManager.swift:25]: L: is `requests.append(request)` an async call so you need a signal?

**Review by marandaneto — APPROVED** [review]
LGTM

File: Tests/SentryTests/Networking/TestRequestManager.swift (hunk 1 above)
Comment (bruno-garcia) [inline Tests/SentryTests/Networking/TestRequestManager.swift:25]: Where is this being read? You need to synchronize the readers too

Comment (philipphofmann) [reply Tests/SentryTests/Networking/TestRequestManager.swift:25]: Good point. Thanks @bruno-garcia

Comment (philipphofmann) [reply Tests/SentryTests/Networking/TestRequestManager.swift:25]: No, it is not async. `add` can be called from multiple background threads so we need to sync the access to `requests`.

**Pull request merged**